my-app/src/pages/newPostPage/AddPostPage.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {useCallback, useContext, useEffect, useRef, useState} from 'react';
+import {Button, Paper, TextField} from "@mui/material";
+import styled from "styled-components";
+import {NewPostContext} from "./NewPostContext";
+import "easymde/dist/easymde.min.css";
+import SimpleMdeReact from "react-simplemde-editor";
+
+
+export const AddPostPage = () => {
+    const logic = useContext(NewPostContext)
+    const inputFileRef = useRef<HTMLInputElement | null>(null)
+
+    return (
+        <Wrapper style={{padding: 30}}>
+            <Button onClick={() => inputFileRef?.current?.click()} variant="outlined" size="large">
+                Upload preview
+            </Button>
+            <input ref={inputFileRef} type="file" onChange={(e) => logic.uploadFileHandler(e)} hidden/>
+            {logic.imageUrl && (
+                <>
+                    <Button variant="contained" color="error" onClick={() => logic.removeFileHandler()}>
+                        Delete
+                    </Button>
+                    <ImageWrapper src={`http://localhost:3222${logic.imageUrl}`} alt="Uploaded"/>
+                </>
+            )}
+
+            <br/>

```

Comment:
use css for spacing, this won't look good in your portfolio

Reply:
deleted it